python/pyspark/sql/tests/test_arrow.py

```diff
@@ -415,6 +415,20 @@ def run_test(num_records, num_parts, max_records, use_delay=False):
         for case in cases:
             run_test(*case)
 
+    def test_createDateFrame_with_category_type(self):
+        pdf = pd.DataFrame({"A": [u"a", u"b", u"c", u"a"]})
+        pdf["B"] = pdf["A"].astype('category')
+
+        with self.sql_conf({"spark.sql.execution.arrow.pyspark.enabled": True}):
+            arrow_df = self.spark.createDataFrame(pdf)
+            result_arrow = arrow_df.toPandas()
+
+        with self.sql_conf({"spark.sql.execution.arrow.pyspark.enabled": False}):
+            df = self.spark.createDataFrame(pdf)
+            result_spark = df.toPandas()
+
+        assert_frame_equal(result_spark, result_arrow)
+
```

Comment:
could you add an assert that the Spark DataFrame has column "B" as a string type?

Reply:
Done, move other test checks here too.